src/acl/DomainData.cc

```diff
@@ -78,23 +78,23 @@ aclDomainCompare(T const &a, T const &b)
             // When a.example.com comes after .example.com in an ACL
             // sub-domain is ignored. That is okay. Just important
             bool d3big = (strlen(d3) > strlen(d4)); // Always suggest removing the longer one.
-            debugs(28, DBG_IMPORTANT, "WARNING: '" << (d3big?d3:d4) << "' is a subdomain of '" << (d3big?d4:d3) << "'");
-            debugs(28, DBG_IMPORTANT, "WARNING: You should remove '" << (d3big?d3:d4) << "' from the ACL named '" << AclMatchedName << "'");
+            debugs(28, DBG_IMPORTANT, "WARNING: '" << (d3big?d3:d4) << "' is a subdomain of '" << (d3big?d4:d3) << "'" <<
+                    Debug::Extra <<  "you should remove '" << (d3big?d3:d4) << "' from the ACL");
             debugs(28, 2, "Ignore '" << d3 << "' to keep splay tree searching predictable");
         }
     } else if (ret == 0) {
         // It may be an exact duplicate. No problem. Just drop.
         if (strcmp(d1,d2)==0) {
-            debugs(28, 2, "WARNING: '" << d2 << "' is duplicated in the list.");
-            debugs(28, 2, "WARNING: You should remove one '" << d2 << "' from the ACL named '" << AclMatchedName << "'");
+            debugs(28, 2, "WARNING: '" << d2 << "' is duplicated in the list." <<
+                    Debug::Extra << "you should remove one '" << d2 << "' from the ACL");
             return ret;
         }
         // When a.example.com comes before .example.com in an ACL
         // discarding the wildcard is critically bad.
         // or Maybe even both are wildcards. Things are very weird in those cases.
         bool d1big = (strlen(d1) > strlen(d2)); // Always suggest removing the longer one.
-        debugs(28, DBG_CRITICAL, "ERROR: '" << (d1big?d1:d2) << "' is a subdomain of '" << (d1big?d2:d1) << "'");
-        debugs(28, DBG_CRITICAL, "ERROR: You need to remove '" << (d1big?d1:d2) << "' from the ACL named '" << AclMatchedName << "'");
+        debugs(28, DBG_CRITICAL, "ERROR: '" << (d1big?d1:d2) << "' is a subdomain of '" << (d1big?d2:d1) << "'" <<
+                Debug::Extra << "you need to remove '" << (d1big?d1:d2) << "' from the ACL");
         self_destruct();
     }
```

Comment:
Please try to add test-suite/squidconf test cases for all three primary problematic outcomes this PR deals with:

1. A level-1 WARNING,
2. A level-2 WARNING (that you might still be a able to test for by setting debug_options to `ALL,1 28,2`), and
3. A fatal ERROR.

Reply:
Done for this and other two cases (in HttpStatus.cc and Ip.cc).

>  setting debug_options to ALL,1 28,2,

No, this does not work: the "-k parse" ignores these settings. I added DBG_PARSE_NOTE() instead.